replication/row_event.go

```diff
@@ -730,6 +740,22 @@ func (e *TableMapEvent) GeometryTypeMap() map[int]uint64 {
 	return ret
 }
 
+// VisibilityMap returns a map: column index -> visiblity.
+// Invisible column was introduced in MySQL 8.0.23
+// nil is returned if not available.
+func (e *TableMapEvent) VisibilityMap() map[int]bool {
+	if len(e.VisibilityBitmap) == 0 {
+		return nil
+	}
+	p := 0
+	ret := make(map[int]bool)
+	for i := 0; i < int(e.ColumnCount); i++ {
+		ret[i] = e.VisibilityBitmap[p/8]&(1<<uint(7-p%8)) != 0
```

Comment:
nit: Maybe it's a little easier to understand the [original code](https://github.com/mysql/mysql-server/blob/ea1efa9822d81044b726aab20c857d5e1b7e046a/libbinlogevents/src/rows_event.cpp#L316) with countdown instead of `1<<uint(7-p%8)`

Reply:
@atercattus 
I appreciate your review. Also, I agree with you and I can get this fixed.
Actually [`VisibilityMap`](https://github.com/dongwook-chan/go-mysql/blob/ccabd266aaa188de9046ee782c144fec36fc7dab/replication/row_event.go#L743-L757) is copy and paste of existing [`UnsignedMap()`](https://github.com/dongwook-chan/go-mysql/blob/ccabd266aaa188de9046ee782c144fec36fc7dab/replication/row_event.go#L614-L631). Should I fix [`UnsignedMap()`](https://github.com/dongwook-chan/go-mysql/blob/ccabd266aaa188de9046ee782c144fec36fc7dab/replication/row_event.go#L614-L631) too?

Reply:
Oh. @lance6716 what do you think about this? Apply the Boy Scout Rule or not?

Reply:
If we decide to improve, the code will look like below.
```go
func (e *TableMapEvent) VisibilityMap() map[int]bool {
	if len(e.VisibilityBitmap) == 0 {
		return nil
	}
	ret := make(map[int]bool)
	for _, field := range e.VisibilityBitmap {
		for c, i := 0x80, 0; c != 0; c, i = c>>1, i+1 {
			ret[i] = field&byte(c) != 0
		}
	}
	return ret
}
```

Reply:
The old code is really not easy to understand 😢 I also hope we start to improve the readability. As for UnsignedMap, I'm OK that you also modify it in this PR or another PR. WDYT @atercattus 

Your new code snippet seems not optimal. The status of `i` should be kept when iterate bytes of `VisibilityBitmap`. And we can early stop when the column count is not a multiple of 8 (though the [original code](https://github.com/mysql/mysql-server/blob/ea1efa9822d81044b726aab20c857d5e1b7e046a/libbinlogevents/src/rows_event.cpp#L316) seems not optimize about this). Maybe we can

```
...
i := 0
for e.VisibilityBitmap {
  for c := 0x80; ... {
    ret[i] = ...
    i++
    if i >= e.ColumnCount { return }
  }
}
...
```
Welcome to find different code

Reply:
My bad. `i` should indeed be declared outside of the inner loop.